tests/suite/scale_test.go

```diff
@@ -0,0 +1,817 @@
+				url = fmt.Sprintf("%s://%d.example.com:%d", protocol, i, portFwdPort)
+			} else if protocol == "https" && portFwdHTTPSPort != 0 {
+				url = fmt.Sprintf("%s://%d.example.com:%d", protocol, i, portFwdHTTPSPort)
+			}
+
+			startCheck := time.Now()
+
+			Eventually(
+				createResponseChecker(url, address),
+			).WithTimeout(30 * time.Second).WithPolling(100 * time.Millisecond).Should(Succeed())
+
+			ttr := time.Since(startCheck)
+
+			seconds := ttr.Seconds()
+			record := []string{strconv.Itoa(i + 1), strconv.FormatFloat(seconds, 'f', -1, 64)}
+
+			Expect(writer.Write(record)).To(Succeed())
+		}
+
+		writer.Flush()
+		Expect(ttrCsvFile.Close()).To(Succeed())
+
+		Expect(
+			framework.GenerateTTRPNG(testResultsDir, ttrCsvFile.Name(), "ttr.png"),
+		).To(Succeed())
+
+		Expect(os.Remove(ttrCsvFile.Name())).To(Succeed())
+
+		Expect(resourceManager.Delete(objects.BaseObjects)).To(Succeed())
+		for i := 0; i < len(objects.ScaleIterationGroups); i++ {
+			Expect(resourceManager.Delete(objects.ScaleIterationGroups[i])).To(Succeed())
+		}
+	}
+
+	runScaleUpstreams := func() {
+		Expect(resourceManager.ApplyFromFiles(upstreamsManifests, ns.Name)).To(Succeed())
+		Expect(resourceManager.WaitForAppsToBeReady(ns.Name)).To(Succeed())
+
+		url := "http://hello.example.com"
+		if portFwdPort != 0 {
+			url = fmt.Sprintf("http://hello.example.com:%d", portFwdPort)
+		}
+
+		Eventually(
+			createResponseChecker(url, address),
+		).WithTimeout(5 * time.Second).WithPolling(100 * time.Millisecond).Should(Succeed())
+
+		Expect(
+			resourceManager.ScaleDeployment(ns.Name, "backend", upstreamServerCount),
+		).To(Succeed())
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+		Expect(resourceManager.WaitForPodsToBeReady(ctx, ns.Name)).To(Succeed())
+
+		Eventually(
+			createResponseChecker(url, address),
+		).WithTimeout(5 * time.Second).WithPolling(100 * time.Millisecond).Should(Succeed())
+
+		Expect(
+			resourceManager.DeleteFromFiles(upstreamsManifests, ns.Name),
+		).To(Succeed())
+	}
+
+	setNamespace := func(objects framework.ScaleObjects) {
+		for _, obj := range objects.BaseObjects {
+			obj.SetNamespace(ns.Name)
+		}
+		for _, objs := range objects.ScaleIterationGroups {
+			for _, obj := range objs {
+				obj.SetNamespace(ns.Name)
+			}
+		}
+	}
+
+	It(fmt.Sprintf("scales HTTP listeners to %d", httpListenerCount), func() {
+		const testName = "TestScale_Listeners"
+
+		testResultsDir := filepath.Join(resultsDir, testName)
+		Expect(os.MkdirAll(testResultsDir, 0755)).To(Succeed())
+
+		objects, err := framework.GenerateScaleListenerObjects(httpListenerCount, false /*non-tls*/)
+		Expect(err).ToNot(HaveOccurred())
+
+		setNamespace(objects)
+
+		runTestWithMetricsAndLogs(
+			testName,
+			testResultsDir,
+			func() {
+				runScaleResources(
+					objects,
+					testResultsDir,
+					"http",
+				)
+			},
+		)
+
+		Expect(
+			os.RemoveAll(filepath.Join(testResultsDir, "manifests")),
+		).To(Succeed())
```

Comment:
This should go in an `AfterEach` for all tests.

Reply:
this is actually no longer necessary, since the tests don't create manifests directory. removing